src/main/java/game/PerftTest.java

```diff
@@ -1,20 +1,19 @@
 package org.lichess.compression.game;
 
-import java.util.ArrayList;
-
 public class PerftTest {
     public static long perft(Board board, int depth) {
         // Counts the number of paths in the tree of legal moves. Useful for
         // testing performance and comparing to well known numbers.
         if (depth >= 1) {
-            ArrayList<Move> moves = new ArrayList<Move>(255);
+            MoveList moves = new MoveList();
```

Comment:
Given the importance of ordering elements, I wonder if `SortedSet` (such as a `TreeSet`) would be overkill.